lib/valueflow.cpp

```diff
@@ -4453,7 +4453,7 @@ struct ConditionHandler {
                     if (!Token::Match(tok, "%assign%|++|--") &&
                         findExpression(cond.vartok->exprId(), start, end, [&](const Token* tok2) {
                         return Token::Match(tok2->astParent(), "%assign%") && astIsLHS(tok2);
-                    })) {
+                    }) && !Token::findmatch(start, "break|continue|goto|throw", end)) {
```

Comment:
Ideally we should check that `break|continue|return` refer to the loop in question.